x-pack/platform/plugins/shared/fleet/server/tasks/bulk_upgrade_packages_task.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { DEFAULT_SPACE_ID } from '@kbn/spaces-plugin/common';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import type { ElasticsearchClient, Logger, SavedObjectsClientContract } from '@kbn/core/server';
+
+import { HTTPAuthorizationHeader } from '../../common/http_authorization_header';
+import { installPackage } from '../services/epm/packages';
+import { appContextService, packagePolicyService } from '../services';
+import { PACKAGE_POLICY_SAVED_OBJECT_TYPE, SO_SEARCH_LIMIT } from '../constants';
+
+const TASK_TYPE = 'fleet:packages-bulk-upgrade';
```

Comment:
I am wondering if we should have a generic task type for all package bulk operations, instead of a specific one for each, upgrade, uninstall